src/walk.rs

```diff
@@ -178,7 +181,7 @@ pub fn scan(root: &Path, pattern: Arc<Regex>, config: Arc<FdOptions>) {
 
             let entry_path = entry.path();
 
-            if entry_path == root {
+            if entry.depth() == 0 {
```

Comment:
I like the change, but does this change something semantically?

Reply:
This shouldn't change anything in practice, this is logically equivalent to the previous if statement. We are just using a different property of root paths to determine this is a root path. The root path is always a depth of zero so we can use that instead checking if entry shares a name with one of the provided paths.